pkg/utils/utils.go

```diff
@@ -106,9 +104,21 @@ func RetrieveVcConfigSecret(params map[string]interface{}, config *rest.Config,
 		logger.WithError(err).Errorf("Failed to get k8s secret, %s", vsphere_secrets)
 		return err
 	}
+	// No kv pairs in the secret.
+	if len(secret.Data) == 0 {
+		errMsg := fmt.Sprintf("Failed to get k8s secret, %s", vsphere_secrets)
+		logger.Errorf(errMsg)
+		return errors.New(errMsg)
+	}
 
-	sEnc := string(secret.Data[secretData])
-	lines := strings.Split(sEnc, "\n")
+	var sEnc string
+	var lines []string
+	for _, value := range secret.Data {
```

Comment:
If there are more than one entry of secret data(possibly from some unexpected configuration by cluster admin), we are going to always use the last one. Do we plan to call it out?

Reply:
We are always going to use the first one, I am not sure if we want to call it out explicitly, especially since it's clearly called out in the csi docs (https://vsphere-csi-driver.sigs.k8s.io/driver-deployment/installation.html)